### PR TITLE
Paypal redirect user

### PR DESCRIPTION
**Pull request by jeevitha90**
Completed User Stories

1. US61701 - Magento 2: PayPal Redirect User
2. US61702 - Magento 2: PayPal Display Confirmation Screen
3. US61703 - Magento 2: PayPal Confirmation Screen

### CODE REVIEW

File: HPS/Heartland/Block/Orderreview.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * @copyright   Heartland (http://heartland.us)
+ * @license     https://github.com/hps/heartland-magento2-extension/blob/master/LICENSE.md
+ */
+namespace HPS\Heartland\Block;
+ 
+class Orderreview extends \Magento\Framework\View\Element\Template
+{
+    public function getOrderId()
+    {       
+		return $_GET['oid'];
```
Comment (SenthilKumar-86) [inline HPS/Heartland/Block/Orderreview.php]: format needs to be fixed in all the files

File: HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml [hunk 2]
```diff
@@ -0,0 +1,124 @@
+<?php
+$lid = $this->getOrderID();
+		 
+$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+$order = $objectManager->create('Magento\Sales\Model\Order')->load($lid);
```
Comment (SenthilKumar-86) [inline HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml]: This code needs to be moved to block. And avoid using object manager directly use dependency injection

File: HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+			$items = $order->getAllItems();
+			foreach($items as $i):			
+				$_product = $objectManager->create('Magento\Catalog\Model\Product')->load($i->getProductId());	
+			//	echo '<pre>'; print_r($_product->debug()); echo '</pre>';
+					$_productName = $_product->getName();
+					$_productSku = $_product->getSku();						
+					$_productPrice = $_product->getPrice();							
+					$_productQty = intval($i->getQtyOrdered());		
+					
+				echo "<tr>";
```
Comment (SenthilKumar-86) [inline HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml]: avoid echo to output html code. use echo only for the PHP variables

File: HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+              <th width="10%">Quantity</th>
+              <th>Subtotal</th>
+          </tr>
+      </thead>
+      <tbody>
+          <?php 			     
+			// Product Items
+			$items = $order->getAllItems();
+			foreach($items as $i):			
+				$_product = $objectManager->create('Magento\Catalog\Model\Product')->load($i->getProductId());
```
Comment (SenthilKumar-86) [inline HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml]: this code can be moved to block

File: HPS/Heartland/Block/Orderreview.php (hunk 1 above)
Comment (jeevitha90) [reply HPS/Heartland/Block/Orderreview.php]: Format has been fixed in all the files

File: HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml (hunk 2 above)
Comment (jeevitha90) [reply HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml]: Yes Code has been moved to block.

File: HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml (hunk 3 above)
Comment (jeevitha90) [reply HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml]: Fixed

File: HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml (hunk 4 above)
Comment (jeevitha90) [reply HPS/Heartland/view/frontend/templates/paypal/review/orderreview.phtml]: Fixed

File: HPS/Heartland/etc/frontend/routes.xml [hunk 5]
```diff
@@ -20,5 +20,14 @@
...
         </route>
+        <route id="orderreview" frontName="hpsorder">
+            <module name="HPS_Heartland" />
+        </route>
+        <route id="cancelorder" frontName="hpsordercancel">
+            <module name="HPS_Heartland" />
+        </route>
+        <route id="completeorder" frontName="hpsordercomplete">
+            <module name="HPS_Heartland" />
+        </route>
```
Comment (slogsdon) [inline HPS/Heartland/etc/frontend/routes.xml:31]: These shouldn't be their own frontnames. They should fall under the `hpspaypal` frontname.

To be honest, we should only have one frontname, `heartland`, but that can be an improvement outside of this PR.

**Review by slogsdon — CHANGES_REQUESTED** [review]

**Comment by slogsdon** [comment]
Closing due to changes in PayPal's policies

**Pull request closed without merging**